packages/frontend/app/services/messages/factory.ts

```diff
@@ -58,6 +60,32 @@ export default class MessageFactory extends Service {
     return message;
   }
 
+  buildChannelVote(vote: Vote, to: Channel) {
+    return this.build({
+      type: TYPE.CHANNEL_VOTE,
+      to: to.id,
+      metadata: buildVote(vote),
+      channelInfo: buildChannelInfo(to),
+    });
+  }
+
+  buildChannelInfoSyncRequest(to: Channel) {
+    return this.build({
+      type: TYPE.INFO_CHANNEL_SYNC,
+      to: to.id,
+      channelInfo: buildChannelInfo(to),
+    });
+  }
+
+  buildChannelInfoSyncFulfill(to: Identity, data: Channel) {
+    return this.build({
+      type: TYPE.INFO_CHANNEL_SYNC,
+      to: to.uid,
+      body: 'TODO make this better', //TODO decide if there is a better way to indicate request vs fulfill
```

Comment:
you could do two different types?

Reply:
That would probably make it cleaner. We'll pursue that.

Reply:
Addressed in 94dc994. This will require some change to #669